packages/compiler/src/core/checker.ts

```diff
@@ -830,6 +870,28 @@ export function createChecker(program: Program): Checker {
     return type;
   }
 
+  /**
+   * Check and resolve a type for the given type reference node, returning its symbol
+   * and type.
+   * @param node Node.
+   * @param mapper Type mapper for template instantiation context.
+   * @param instantiateTemplate If templated type should be instantiated if they haven't yet.
+   * @returns Resolved type.
+   */
+  function getTypeReferenceTypeAndSymbol(
+    node: TypeReferenceNode | MemberExpressionNode | IdentifierNode,
+    mapper: TypeMapper | undefined,
+    instantiateTemplate = true
+  ): { sym: Sym | null; type: Type } {
```

Comment:
i never fully get when we should use `null`, always try to stick to `undefined` so you don't have to check for multiple things

Reply:
I tend to use null for an explicit no thing value since undefined crops up randomly, but do note that much of this code will be completely rewritten once the resolver is operational.